### PR TITLE
Cursed heart revamp.

### PR DESCRIPTION
**Pull request by orthography**
## About The Pull Request

~- Wizard cursed heart variant added back (was it even there from the start?) into the wizard shop for **1 point.**~

~- Necropolis chests now spawn the correct version of cursed heart as specified in the wiki.~

~- Wizard cursed heart has been given it's own name and description to distinguish it from the normal one.~

~Fuck you miners.~

Cursed heart has been entirely overhauled. Through blood sweat and tears this AFK killer is useful.

The cursed heart is an item that must be pumped every 7 seconds (pausing on DC or ping greater than 250), providing healing. Forgetting to pump the heart will give you a warning and turn your screen red. It will then apply a `cursed` effect, gradually increasing with more severe and debilitating effects. (eventual bloodloss after 14-100 missed pumps). These effects are similar to drunkenness.

Furthermore, not pumping the heart can cause the user to expell their curse in the form of vomit curing them.

There are two variants, wizard and miner. Wizard is minorly improved (extra healing on use).

## Why It's Good For The Game

Cursed heart was a pretty hit-or-miss item of either being used prolifically or totally unused depending on the miner. This PR aims to somewhat match the standard of certain mining items being weaker versions of wizard/antagonist gear. (ie dusty shard not allowing you to pick what holo you get as opposed to tarot cards, summon item book being a spell, miner's soulstone compared to wizard/cult stone, etc)

However, as stated in this very PR, it was a noob killer, or too demanding for the /tg/ playerbase ~presumably picking their noses and or asses~. This seeks to rectify that by making the item more accessible and less of a burden.

**I sincerely hope to see wizards using this!**

https://youtu.be/AgzWlzam51I?t=540

## Changelog

:cl:
add: added a name and description to the wizard cursed heart
balance: miner loot now spawns the correct cursed heart
fix: After extensive deliberation in the Wizard Federation, harvesting of cursed hearts has resumed and are now available from your spellbooks.
spellcheck: changed the text of the wizard cursed heart
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
What’s the point in the mining one, then?

**Comment by Fikou** [comment]
Why is this good for the game?

**Comment by Arkatos1** [comment]
Honestly, I would rather see the heart completely removed or reduced to admin smite only than for it to proliferate in the game further.

**Comment by GuillaumePrata** [comment]
> Honestly, I would rather see the heart completely removed or reduced to admin smite only than for it to proliferate in the game further. Just my 2 cents.

With how it works now it will stay as a noob trap that get people killed or it needs to be a really powerful item to balance it's downsides.

Unless it is reworked a lot, it's definitely better to just kill the item.

**Comment by orthography** [comment]
> > Honestly, I would rather see the heart completely removed or reduced to admin smite only than for it to proliferate in the game further. Just my 2 cents.
> 
> With how it works now it will stay as a noob trap that get people killed or it needs to be a really powerful item to balance it's downsides.
> 
> Unless it is reworked a lot, it's definitely better to just kill the item.

@GuillaumePrata How would you have it changed to be worth the downsides, as it is now it can heal up to 100 damage every 6 seconds (with wizard variant)

**Comment by newfren** [comment]
why are we nerfing the worst chest reward lmao

**Comment by Shadowflame909** [comment]
Cursed Heart is lag/afk killbait the item

Any wizard that gets it should get an extra POINT for playing with a literal timer that drains 50% of their blood every time they dont beat it

**Comment by orthography** [comment]
Changed the stats around after playing a round with all wizards. Seeing as I was able to kill 3 wizards, I firmly believe that it's balanced (with minor improvements). It is not as demanding as people think it is. And it is much more forgiving than people think it is. You'd have to ignore it for ~36 seconds straight to be killed by it.

**Comment by GuillaumePrata** [comment]
> How would you have it changed to be worth the downsides, as it is now it can heal up to 100 damage every 6 seconds (with wizard variant)

I would just change the downside instead of the upside, make it not kill you if you don't beat it and that you have to charge/manually beat your heart for a few seconds to get a momentary boost out of it.

This way it is not a consistent healing as long as you can micro manage it and instead it rewards taking small breaks to let the heal kick in.

As Shadow said, if anything happens with your connection the cursed heart will kill you on our current version and miners/wizards are unlikely to be revived when this happens.

**Comment by JohnFulpWillard** [comment]
Not sure if you plan on implementing any of the comment's suggestions, but currently the heart makes the wizard invincible to everything including stamina damage, as long as they aren't put to sleep or lag out, for one point. Don't know if that's your intent here, but I don't really see this being very balanced, especially for laggy players.

**Comment by orthography** [comment]
@GuillaumePrata @Shadowflame909 @newfren @JohnFulpWillard @Arkatos1 Hope this new version seems more manageable and interesting than bloodloss to instadeath.

File: code/datums/status_effects/debuffs/curse.dm [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+/datum/status_effect/cursed/get_examine_text()
+	// Dead people don't seem cursed
+	if(owner.stat == DEAD || HAS_TRAIT(owner, TRAIT_FAKEDEATH))
+		return null
+
+	switch(curse_value)
+		if(11 to 21)
+			return span_warning("[owner.p_they(TRUE)] [owner.p_are()] slightly pale.")
+		if(21.01 to 41)
```
Comment (Fikou) [inline code/datums/status_effects/debuffs/curse.dm:33]: you dont need this .01 stuff, can just do from 21 to 41

File: code/modules/surgery/organs/heart.dm [hunk 2]
```diff
@@ -107,13 +107,14 @@
...
+	var/blood_regained = 70
 
 	//How much to heal per pump, negative numbers would HURT the player
-	var/heal_brute = 0
-	var/heal_burn = 0
-	var/heal_oxy = 0
+	var/heal_brute = 25
+	var/heal_burn = 25
+	var/heal_oxy = 10
+	var/heal_toxin = 10
```
Comment (Fikou) [inline code/modules/surgery/organs/heart.dm:117]: lets not make miners heal toxin freely

File: code/modules/surgery/organs/heart.dm [hunk 3]
```diff
@@ -107,13 +107,14 @@
...
-	var/blood_loss = 100 //600 blood is human default, so 5 failures (below 122 blood is where humans die because reasons?)
+	var/pump_delay = 70 //you can pump 1 second early, for lag, but no more (otherwise you could spam heal)
+	var/blood_regained = 70
 
 	//How much to heal per pump, negative numbers would HURT the player
-	var/heal_brute = 0
-	var/heal_burn = 0
-	var/heal_oxy = 0
+	var/heal_brute = 25
+	var/heal_burn = 25
```
Comment (Fikou) [inline code/modules/surgery/organs/heart.dm:115]: id lower this to 20

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/heart.dm (hunk 3 above)
Comment (orthography) [reply code/modules/surgery/organs/heart.dm]: https://github.com/tgstation/tgstation/pull/68739/commits/e6fb9393dfc3cb3f2e5c0a1e43fbec85ba2f6066

File: code/modules/surgery/organs/heart.dm (hunk 2 above)
Comment (orthography) [reply code/modules/surgery/organs/heart.dm]: https://github.com/tgstation/tgstation/pull/68739/commits/e6fb9393dfc3cb3f2e5c0a1e43fbec85ba2f6066

File: code/datums/status_effects/debuffs/curse.dm (hunk 1 above)
Comment (orthography) [reply code/datums/status_effects/debuffs/curse.dm]: https://github.com/tgstation/tgstation/pull/68739/commits/e6fb9393dfc3cb3f2e5c0a1e43fbec85ba2f6066

**Comment by orthography** [comment]
@Fikou does the current PR state meet your needs?

File: code/modules/surgery/organs/heart.dm [hunk 4]
```diff
@@ -126,18 +127,18 @@
 
 /obj/item/organ/internal/heart/cursed/on_life(delta_time, times_fired)
 	if(world.time > (last_pump + pump_delay))
-		if(ishuman(owner) && owner.client) //While this entire item exists to make people suffer, they can't control disconnects.
+		if(ishuman(owner) && owner.client && owner.client?.lastping < 250) //While this entire item exists to make people suffer, they can't control disconnects.
```
Comment (optimumtact) [inline code/modules/surgery/organs/heart.dm:130]: my ping is consistently over 250, which means I can use this item without any ill effects whatsoever

this is simply not a workable solution.

**Review by Fikou — APPROVED** [review]
